pkg/ndctl/region.go

```diff
@@ -208,13 +204,36 @@ func (r *region) CreateNamespace(opts CreateNamespaceOpts) (Namespace, error) {
 	}
 
 	if opts.Size != 0 {
-		ways := uint64(C.ndctl_region_get_interleave_ways(r))
-		align = align * ways
-		if opts.Size%align != 0 {
-			// Round up size to align with next block boundary.
-			opts.Size = (opts.Size/align + 1) * align
-			klog.V(4).Infof("%s: namespace size must align to interleave-width:%d * alignment:%d, force-align to %d",
-				regionName, ways, align, opts.Size)
+		/* old method, using namespace align and interleave ways.
+		If we use fixed 2M namespace align, this is always overshadowed
+		by region align which is 16 or 96 MB so we may leave this code out
+		ways := r.InterleaveWays()
+				align = align * ways
+				if opts.Size%align != 0 {
+					// Align down to namespace-alignment boundary.
+					opts.Size /= align
+					opts.Size *= align
+					klog.V(4).Infof("%s: namespace size must align to interleave-width:%d * alignment:%d, adjust to %d",
+						regionName, ways, align, opts.Size)
+				}*/
+		regionalign := r.GetAlign()
+		if opts.Size%regionalign != 0 {
+			// Align down to region-alignment boundary.
+			opts.Size /= regionalign
+			opts.Size *= regionalign
+			klog.V(4).Infof("%s: namespace size must align to region alignment:%d, adjust to %d",
+				regionName, regionalign, opts.Size)
```

Comment:
Instead of modifying opts, can your create a local variable? Then in this info message you can print the original value and the new one.

Reply:
Not quite solved the way I meant (`opts.Size` still gets modified eventually), but the info message is more informative, so let's consider this resolved.